src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/PartialViewResultExecutor.cs

```diff
@@ -0,0 +1,141 @@
+            : base(viewOptions, viewEngine, telemetry)
+        {
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            Logger = logger;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="ILogger"/>.
+        /// </summary>
+        protected ILogger Logger { get; }
+
+        /// <summary>
+        /// Attempts to find the <see cref="IView"/> associated with <paramref name="viewResult"/>.
+        /// </summary>
+        /// <param name="actionContext">The <see cref="ActionContext"/> associated with the current request.</param>
+        /// <param name="viewResult">The <see cref="PartialViewResult"/>.</param>
+        /// <returns>A <see cref="ViewEngineResult"/>.</returns>
+        public virtual ViewEngineResult FindView(ActionContext actionContext, PartialViewResult viewResult)
+        {
+            if (actionContext == null)
+            {
+                throw new ArgumentNullException(nameof(actionContext));
+            }
+
+            if (viewResult == null)
+            {
+                throw new ArgumentNullException(nameof(viewResult));
+            }
+
+            var viewEngine = viewResult.ViewEngine ?? ViewEngine;
+            var viewName = viewResult.ViewName ?? actionContext.ActionDescriptor.Name;
+
+            var result = viewEngine.FindView(actionContext, viewName);
+            if (!result.Success)
+            {
+                if (Telemetry.IsEnabled("Microsoft.AspNet.Mvc.ViewNotFound"))
+                {
+                    Telemetry.WriteTelemetry(
+                        "Microsoft.AspNet.Mvc.ViewNotFound",
+                        new
+                        {
+                            actionContext = actionContext,
+                            isPartial = false,
+                            result = viewResult,
+                            viewName = viewName,
+                            searchedLocations = result.SearchedLocations
+                        });
+                }
+
+                Logger.LogError(
+                    "The partial view '{PartialViewName}' was not found. Searched locations: {SearchedViewLocations}",
+                    viewName,
+                    result.SearchedLocations);
+            }
+            
+            if (Telemetry.IsEnabled("Microsoft.AspNet.Mvc.ViewFound"))
+            {
+                Telemetry.WriteTelemetry(
+                    "Microsoft.AspNet.Mvc.ViewFound",
+                    new
+                    {
+                        actionContext = actionContext,
+                        isPartial = false,
+                        result = viewResult,
+                        viewName,
+                        view = result.View,
+                    });
+            }
+
+            Logger.LogVerbose("The partial view '{PartialViewName}' was found.", viewName);
+
+            return result;
```

Comment:
This code *looks* identical to `ViewResultExecutor`, but it's actually just different enough IMO to deserve its own method.